File: eMovieFinder/eMovieFinder.Model/Dtos/Requests/Actor/ActorUpdateRequest.cs
using System;
using System.ComponentModel;

namespace eMovieFinder.Model.Dtos.Requests.Actor
{
    public class ActorUpdateRequest
    {
        public string? FirstName { get; set; }
        public string? LastName { get; set; }
        public string? ImagePlainText { get; set; }
        [DefaultValue(null)]
        public byte[]? Image { get; set; }
        public DateTime? BirthDate { get; set; }
        public int? CountryId { get; set; }
        public string? IMDbLink { get; set; }
        public string? Biography { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/dtos/requests/actor/actor_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'actor_update_request.g.dart';

@JsonSerializable()
class ActorUpdateRequest {
  final String? firstName;
  final String? lastName;
  final String? imagePlainText;
  final DateTime? birthDate;
  final int? countryId;
  final String? imDbLink;
  final String? biography;

  ActorUpdateRequest({
    this.firstName,
    this.lastName,
    this.imagePlainText,
    this.birthDate,
    this.countryId,
    this.imDbLink,
    this.biography
  });

  factory ActorUpdateRequest.fromJson(Map<String, dynamic> json) => _$ActorUpdateRequestFromJson(json);
  Map<String, dynamic> toJson() => _$ActorUpdateRequestToJson(this);
}